python/graphstorm/gconstruct/remap_result.py

```diff
@@ -318,32 +463,77 @@ def main(args, gs_config_args):
-        # Try to get it from saved prediction config.
-        with open(os.path.join(predict_dir, "result_info.json"),
-                    "r",  encoding='utf-8') as f:
-            info = json.load(f)
-            pred_etypes = [etype.split(",") for etype in info["etypes"]] \
-                    if "etypes" in info else []
-
-    ntypes = \
-        [etype[0] for etype in pred_etypes] + \
+
+    # If pred_ntypes (nodes with prediction results)
+    # is not empty, we need to remap edge prediction results
+    # Note: For distributed SageMaker runs, pred_ntypes must be
+    # provided if node prediction result remap is required,
+    # as result_info.json is only saved by rank0 and
+    # there is no shared file system.
+    if len(pred_ntypes) > 0:
+        for pred_ntype in pred_ntypes:
+            assert os.path.exists(os.path.join(predict_dir, pred_ntype)), \
+                f"prediction results of {pred_ntype} do not exists."
+
+    if with_shared_fs:
+        # Only when shared file system is avaliable,
+        # we will check result_info.json for
+        # pred_etypes and pred_ntypes.
+        # If shared file system is not avaliable
+        # result_info.json is not guaranteed to exist
+        # on each instances. So users must use
+        # --pred-ntypes or --pred-etypes instead.
+        #
+        # In case when both --pred-ntypes or --pred-etypes
+        # are provided while result_info.json is also avaliable,
+        # GraphStorm remaping will follow --pred-ntypes or --pred-etypes
+        # and ignore the result_info.json.
+        if os.path.exists(os.path.join(predict_dir, "result_info.json")):
+            # User does not provide pred_etypes.
+            # Try to get it from saved prediction config.
+            with open(os.path.join(predict_dir, "result_info.json"),
+                        "r",  encoding='utf-8') as f:
+                info = json.load(f)
+                if len(pred_etypes) == 0:
+                    pred_etypes = [list(etype) for etype in info["etypes"]] \
+                        if "etypes" in info else []
+                if len(pred_ntypes) == 0:
+                    pred_ntypes = info["ntypes"] if "ntypes" in info else []
+
+    ntypes = []
+
+    ntypes += [etype[0] for etype in pred_etypes] + \
         [etype[2] for etype in pred_etypes]
+    ntypes += pred_ntypes
+
+    if len(ntypes) == 0:
+        # Nothing to remap
+        logging.warning("Skip remapping edge predictions and node predictions")
+        return
+
     for ntype in set(ntypes):
+        mapping_file = os.path.join(id_mapping_path, ntype + "_id_remap.parquet")
         logging.debug("loading mapping file %s",
-                      os.path.join(id_mapping_path, ntype + "_id_remap.parquet"))
-        id_maps[ntype] = \
-            IdReverseMap(os.path.join(id_mapping_path, ntype + "_id_remap.parquet"))
+                      mapping_file)
+        if os.path.exists(mapping_file):
+            id_maps[ntype] = \
```

Comment:
where is this used?

Reply:
i cannot find the code below that uses `id_map`

Reply:
id_map is a global variable.
it is accessed in worker_remap_node_pred.

Reply:
Here is an short intro of what id_map is
https://github.com/awslabs/graphstorm/pull/578/files#diff-8ed4342d598e284201f07618bf95b5a085b567face2180fa825155738ac53697R43-R50

Reply:
i see. will it be better if we have a class. if we want the child process to inherit the data, we can create a global object of the class. i think this can make the code easier to manage and understand.